extensions/notebook/src/test/book/book.test.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as should from 'should';
+import * as TypeMoq from 'typemoq';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as rimraf from 'rimraf';
+import { BookTreeViewProvider } from '../../book/bookTreeView';
+import { BookTreeItem } from '../../book/bookTreeItem';
+
+describe('BookTreeViewProvider.getChildren', function (): void {
+	const rootFolderPath = path.join(__dirname, 'testBook');
```

Comment:
>__dirname [](start = 34, length = 9)

We should avoid writing stuff to the path as the source. Using something like os.tmpdir() is preferred.

I would also suggest creating a new subfolder in each run that has a randomly generated name. That way parallel runs aren't trying to write/read from the same place

Reply:
👍 We should always write to temp folder - should be other examples of this elsewhere.

Reply:
changed __dirname to os.tmpdir(). Thanks!